### PR TITLE
Feich/add some general intents

### PR DESCRIPTION
**Pull request by feich-ms**
Add Next/Previous handling logic
## Description
1. Add Next.Previous handling logic for three bots
2. Remove unused code in SkillDialog of each skill
3. Update appsetting config

## Related Issue
#264

## Testing Steps
Launch a skill and ask "show more" if last command is showing to dos/emails/calendar events. You should get expected results instead of "didn't get what you said".

### CODE REVIEW

**Review by bobokids — COMMENTED** [review]
using outlook as default provider?

**Comment by feich-ms** [comment]
@bobokids The VirtualAssistant is using Outlook, so for local mode, I changed it to Outlook to keep the consistence. IPA team can change it to OneNote. It's just a string change in config.

**Comment by darrenj** [comment]
Makes sense to me.

Sent from my 📱
________________________________

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/appsettings.json [hunk 1]
```diff
@@ -4,10 +4,7 @@
   "ApplicationInsights": {
     "InstrumentationKey": ""
   },
-  "SupportedProviders": [
```
Comment (lauren-mills) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/appsettings.json]: Why did you remove these?

**Comment by lauren-mills** [comment]
@feich-ms you will not be able to manually change the todo.luis file going forward. Please see my email on how to update luis models using the .lu files.

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/appsettings.json (hunk 1 above)
Comment (feich-ms) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/appsettings.json]: When I run local mode, here should be connection name defined by users instead of provider name(Azure Active Directory v2). It will bring confusions to users so I removed them. Any insights?

**Comment by feich-ms** [comment]
@lauren-mills thanks to Laruen for letting me know. I will update the luis model soon after the LU model is finalized. For this PR, no LU file changes are needed as these intents(Next/Previous) are already there in General model. I just use them in each skill.

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/appsettings.json (hunk 1 above)
Comment (lauren-mills) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/appsettings.json]: In the multiple authentication providers PR I redesigned how this works. Authentication connections are defined in the .bot file. The "SupportedProvider" field in appsettings is to show which providers are supported by the skill (i.e. The CalendarSkill supports Microsoft and Google calendars.) The SkillConfiguration class uses these to filter out unsupported providers from the login prompt. For example, if a bot is configured for Microsoft, Google, and Github accounts, but a skill only supports Microsoft and Google, only Microsoft and Google will be shown as options when they login.

Comment (feich-ms) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/appsettings.json]: I know the design in skill mode. But in local mode, current implementation didn't work. Please take a look at below implementations of current SkillConfiguration.cs. It seems they are not aligned with your changes in BotServices.cs. I made a change https://github.com/Microsoft/AI/pull/265/commits/14ab98ab9b4c857a5bec64efdd29d064b8d83e05 here based on your implementations in BotServices.cs, please see if this is the right logic.

                    case ServiceTypes.Generic:
                        {
                            if (service.Name == "Authentication")
                            {
                                var auth = service as GenericService;

                                foreach (var provider in supportedProviders)
                                {
                                    auth.Configuration.TryGetValue(provider, out var connectionName);

                                    if (connectionName != null)
                                    {
                                        AuthenticationConnections.Add(provider, connectionName);
                                    }
                                }
                            }

                            break;
                        }

**Comment by DingmaomaoBJTU** [comment]
>         public const string LocalModeAuth = "LocalAuth";

Can this be removed?

---
Refers to: solutions/Virtual-Assistant/src/csharp/skills/todoskill/Dialogs/Shared/ToDoSkillDialog.cs:29 in 29e56cd. [](commit_id = 29e56cd4a292dbb41a3a46b199e27dc8b642247c, deletion_comment = False)

**Review by DingmaomaoBJTU — APPROVED** [review]
:shipit:

**Comment by feich-ms** [comment]
> > ```
> >     public const string LocalModeAuth = "LocalAuth";
> > ```
> 
> Can this be removed?
> 
> Refers to: solutions/Virtual-Assistant/src/csharp/skills/todoskill/Dialogs/Shared/ToDoSkillDialog.cs:29 in [29e56cd](https://github.com/Microsoft/AI/commit/29e56cd4a292dbb41a3a46b199e27dc8b642247c). [](commit_id = [29e56cd](https://github.com/Microsoft/AI/commit/29e56cd4a292dbb41a3a46b199e27dc8b642247c), deletion_comment = False)

Sure. Removed this for all the three skills

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Skills/SkillConfiguration.cs [hunk 2]
```diff
@@ -50,11 +50,11 @@ public SkillConfiguration(BotConfiguration botConfiguration, string[] supportedP
 
                                 foreach (var provider in supportedProviders)
                                 {
-                                    auth.Configuration.TryGetValue(provider, out var connectionName);
+                                    var matches = auth.Configuration.Where(x => x.Value == provider);
```
Comment (lauren-mills) [inline solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Skills/SkillConfiguration.cs:53]: Yes, @feich-ms this is the right way to check now.

**Review by lauren-mills — APPROVED** [review]
